var/spack/repos/builtin/packages/umpire/package.py

```diff
@@ -458,37 +455,51 @@ def initconfig_package_entries(self):
-            return
-
-        # Run a subset of examples PROVIDED installed
-        # tutorials with readily checkable outputs.
-        checks = {
-            "malloc": ["99 should be 99"],
-            "recipe_dynamic_pool_heuristic": ["in the pool", "releas"],
-            "recipe_no_introspection": ["has allocated", "used"],
-            "strategy_example": ["Available allocators", "HOST"],
-            "tut_copy": ["Copied source data"],
-            "tut_introspection": ["Allocator used is HOST", "size of the allocation"],
-            "tut_memset": ["Set data from HOST"],
-            "tut_move": ["Moved source data", "HOST"],
-            "tut_reallocate": ["Reallocated data"],
-            "vector_allocator": [""],
-        }
-
-        for exe in checks:
-            expected = checks[exe]
-            reason = "test: checking output from {0}".format(exe)
-            self.run_test(
-                exe,
-                [],
-                expected,
-                0,
-                installed=False,
-                purpose=reason,
-                skip_missing=True,
-                work_dir=self.prefix.bin,
-            )
+
+        exe_run = which(join_path(self.prefix.bin, exe))
+        if exe_run is None:
+            raise SkipTest(f"{exe} is not installed for version {self.version}")
+        out = exe_run(output=str.split, error=str.split)
+        check_outputs(expected, out)
+
+    def test_malloc(self):
+        """Run Malloc"""
+        self.run_test("malloc", ["99 should be 99"])
+
+    def test_recipe_dynamic_pool_heuristic(self):
+        """Multiple use allocator test"""
+        self.run_test("recipe_dynamic_pool_heuristic", ["in the pool", "releas"])
+
+    def test_recipe_no_introspection(self):
+        """Test without introspection"""
+        self.run_test("recipe_no_introspection", ["has allocated", "used"])
+
+    def test_strategy_example(self):
+        """Memory allocation strategy test"""
+        self.run_test("strategy_example", ["Available allocators", "HOST"])
+
+    def test_tut_copy(self):
+        """Copy data test"""
+        self.run_test("tut_copy", ["Copied source data"])
+
+    def test_tut_introspection(self):
+        """Keep track of pointer allocation test"""
+        self.run_test("tut_introspection", ["Allocator used is HOST", "size of the allocation"])
+
+    def test_tut_memset(self):
+        """Set entire block of memory to one value test"""
+        self.run_test("tut_memset", ["Set data from HOST"])
+
+    def test_tut_move(self):
+        """Move memory test"""
+        self.run_test("tut_move", ["Moved source data", "HOST"])
+
+    def test_tut_reallocate(self):
+        """Reallocate memory test"""
+        self.run_test("tut_reallocate", ["Reallocated data"])
+
+    def test_vector_allocator(self):
+        """Allocate vector memory test"""
+        self.run_test("vector_allocator", [""])
```

Comment:
These need to be calling `self.run_example()`, not deprecated `self.run_test()`.